src/nupic/utils/Random.cpp

```diff
@@ -191,11 +167,17 @@ UInt32 Random::getUInt32(const UInt32 max) {
   return sample % max;
 }
 
-UInt64 Random::getUInt64(const UInt64 max) {
-  NTA_CHECK(max > 0);
-  const UInt64 lo = impl_->getUInt32();
-  const UInt64 hi = impl_->getUInt32();
-  const UInt64 sample = lo | (hi << 32);
+UInt64 Random::getUInt64(const UInt64 max)
+{
+  NTA_ASSERT(max > 0);
+  UInt64 smax = Random::MAX64 - (Random::MAX64 % max);
+  UInt64 sample, lo, hi;
+  do {
+    lo = impl_->getUInt32();
+    hi = impl_->getUInt32();
+    sample = lo | (hi << 32);
+  } while(sample > smax);
+  // NTA_WARN << "Random64(" << max << ") -> " << sample % max << " smax = " << smax;
```

Comment:
I think this is mistake in merge? Please confirm with community/master and revert to the original (with NTA_CHECK) implementation if it's the case.